src/popover/popover.js

```diff
@@ -474,8 +474,9 @@ class PopoverInner extends React.Component<
                 disabled={!this.props.focusLock}
                 noFocusGuards={false}
                 // see popover-focus-loop.scenario.js for why hover cannot return focus
-                returnFocus={this.props.returnFocus && !this.isHoverTrigger()}
+                returnFocus={!this.isHoverTrigger() && this.props.returnFocus}
```

Comment:
yes, earlier returnFocus was only `boolean`, but now it can be `boolean | FocusOptions | Function`, by changing the order if `this.props.returnFocus` is a function we would pass a function to `returnFocus`.